src/main/java/org/veupathdb/service/eda/ds/core/AbstractPlugin.java

```diff
@@ -334,6 +336,38 @@ protected Map<String, Double> getCategoricalProportionDistribution(VariableSpec
     return response.getHistogram().stream().collect(Collectors.toMap(HistogramBin::getBinLabel, bin -> Double.valueOf(bin.getValue().toString())));
   }
 
+  protected Map<String, InputStream> getVocabByRootEntity(DynamicDataSpecImpl dataSpec, List<APIFilter> subsetFilters) {
+    PluginUtil util = getUtil();
+    ResponseFuture response = _subsettingClient.getVocabByRootEntity(_referenceMetadata, dataSpec, subsetFilters);
+
+    Map<String, InputStream> vocab = new HashMap<String, InputStream>();
+    try (InputStream vocabStream = response.getInputStream()) {
```

Comment:
I think the problem is that when you assign a closeable variable like an `InputStream` in a "try-with-resources" block like this, it closes the stream at the end of the try block.